### ISSUE TITLE
citation-generator-improvement

### ISSUE THREAD

**Issue by zoometh** [eamena-project/eamena-arches-dev#162]
TODO list to improve the citation plugin (https://database.eamena.org/citations/)

**Comment by zoometh** [comment]
- [ ] The '**Advanced Search**' button should be renamed to '**Additional outputs**' and these additional outputs should be documented here: https://github.com/eamena-project/eamena-arches-dev/tree/main/dbs/database.eamena/citation
![image](https://github.com/user-attachments/assets/48c7f576-73fb-416b-b241-f3bfcfc9edfb)

**Comment by zoometh** [comment]
- [ ] improve the `level 2` of maps in `zenodo_map()`, file `zenodo.py`

The management of level 2 in the map function, can probably be made more efficient. Currently, it tries to match a country name with a GeoJSON file hosted here: https://github.com/eamena-project/eamena-data/tree/main/reference_data/countries. For example, these records are creating errors

![image](https://github.com/user-attachments/assets/00018a97-e3e0-4486-acdb-d0fc7dec4226)

and

![image](https://github.com/user-attachments/assets/9c70dcc3-06f5-4adf-8bc6-ec6612eab7d9)

**Comment by zoometh** [comment]
- [ ] set up error controls in `zenodo.py`

More error controls (`try`, `catch`, `break`, etc.) should be set up to handle NA data, typos, etc. to avoid error. For example: 

![image](https://github.com/user-attachments/assets/1f67c5f0-ce7c-498a-8e08-9d2e608a3d39)
![image](https://github.com/user-attachments/assets/442fd5f6-eedc-46f1-bc92-499aca5e7f91)

**Comment by zoometh** [comment]
- [ ] make the success message clickable (hyperlink)

![image](https://github.com/user-attachments/assets/f1ddaca0-2730-480a-92fa-2210efdb5469)

**Comment by reubenosborne1** [comment]
issue 1 should be a simple change of the form code from Advanced Search to Additional outputs.

issue 2 is more complicated as we will have to standardise the naming of the files and countries. The obvious way is to make sure the data is correct in the database. Where is "Jordan ," coming from? Can the data be empty?
For a code fix: i'd recommend renaming all the files to lower case with no spaces. So "Saudi Arabia.geojson" becomes "saudiarabia.geojson".  Then I can create some code in zenedo.py that standardises the Country name. I can strip out spaces and commas. So "Saudi Arabia ," would become saudiarabia and therefore match the file. 
We can also skip a record if the country is empty in the data.

issue 3 is another HTML change.

**Comment by zoometh** [comment]
- [x] The 'Advanced Search' button should be renamed to 'Additional outputs' and these additional outputs should be documented here: https://github.com/eamena-project/eamena-arches-dev/tree/main/dbs/database.eamena/citation
 - [ ] improve the `level 2` of maps in `zenodo_map()`, file `zenodo.py`
 - [ ] set up error controls in zenodo.py
 - [x] make the success message clickable (hyperlink)

**Comment by reubenosborne1** [comment]
Yes everything but the error controls (issue 4). Do you have an example of a GEOJSON url I can use for this @zoometh ?

**Comment by zoometh** [comment]
Which issue in https://github.com/eamena-project/eamena-arches-dev/issues/162#issuecomment-2536266865?

**Comment by reubenosborne1** [comment]
@zoometh issue 3 " set up error controls in zenodo.py"

**Comment by reubenosborne1** [comment]
That geojson worked here: https://sandbox.zenodo.org/records/142230
@zoometh .
